component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -465,6 +467,11 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                 Object verifiedEmailObject = context.getProperty(EmailOTPAuthenticatorConstants.REQUESTED_USER_EMAIL);
                 if (verifiedEmailObject != null) {
                     try {
+                        if (Utils.getThreadLocalToSkipSendingEmailVerificationOnUpdate() != null) {
```

Comment:
If we are setting the value immediately next, no need to unset this.

Reply:
Addressed with https://github.com/wso2-extensions/identity-outbound-auth-email-otp/pull/176/commits/640eaf44ca066d1bb33020b29f94f240b3fb49f5